lib/pages/admin/add_choice_board_category.dart

```diff
@@ -0,0 +1,237 @@
+                            : Icon(
+                                Icons.image_search_outlined,
+                                size: 160,
+                              )),
+                    // instructions text
+                    Text(
+                      "Pick the main category image",
+                      style:
+                          TextStyle(fontSize: 32, fontWeight: FontWeight.bold),
+                    ),
+                    const SizedBox(height: 20),
+                    // buttons to take/upload images
+                    PickImageButton(
+                        label: Text("Choose from Gallery"),
+                        icon: Icon(Icons.image),
+                        onPressed: () =>
+                            pickImage(source: ImageSource.gallery)),
+                    PickImageButton(
+                        label: Text("Take a Picture"),
+                        icon: Icon(Icons.camera_alt),
+                        onPressed: () => pickImage(source: ImageSource.camera)),
+                    const SizedBox(height: 20),
+                    // field to enter the item name
+                    TextField(
+                      controller: categoryNameController,
+                      decoration: InputDecoration(
+                          hintText: "Category's name",
+                          border: InputBorder.none,
+                          hintStyle: TextStyle(fontWeight: FontWeight.bold)),
+                      cursorColor: Colors.white,
+                    ),
+                    const Divider(
+                      color: Colors.black38,
+                      thickness: 0.2,
+                    ),
+                    const SizedBox(height: 20),
+                    // submit to database button
+                    TextButton.icon(
+                        key: const Key("createCategoryButton"),
+                        onPressed: () => saveCategory(
+                            image: selectedImage,
+                            categoryName: categoryNameController.text),
+                        icon: Icon(Icons.add),
+                        label: const Text("Create new category"),
+                        style: const ButtonStyle(
+                            foregroundColor:
+                                MaterialStatePropertyAll(Colors.white),
+                            backgroundColor: MaterialStatePropertyAll(
+                                Color.fromARGB(255, 105, 187, 123))))
+                  ],
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  /// Given a category's image and name,
+  /// - upload the image to the cloud storage
+  /// - create a new category with the uploaded image's Url in Firestore
+  /// - Take the user back to the Choice Boards page
+  void saveCategory(
+      {required File? image, required String? categoryName}) async {
+    if (categoryName!.isEmpty || image == null) {
+      showDialog(
```

Comment:
U can create a method to show the dialog to avoid code duplication. Just create a method with the showDialog() thingy and call it with a different string each time.

-> I did that in the edit account page. Would be even better if you can reuse the method I built somehow by placing it in its all widget file or finding another nicely structured way to use the same method.

Reply:
Thanks for the suggestion, though I think this is a needless complication as the amount of written code is the same in practice, if not more (function name: `showDialog()` vs `AlertDialog()`, parameters: `dialogText` vs `content`, etc). 

If you feel particularly strong about this I am absolutely happy to change it!